ts/components/Pinpad/index.tsx

```diff
@@ -0,0 +1,107 @@
+import { fromNullable } from "fp-ts/lib/Option";
+import { View } from "native-base";
+import * as React from "react";
+import { TextInput, TouchableOpacity } from "react-native";
+
+import { PinString } from "../../types/PinString";
+import { PIN_LENGTH } from "../../utils/constants";
+
+import { styles } from "./Pinpad.style";
+import { Baseline, Bullet } from "./Placeholders";
+
+const fucusElement = (el: TextInput) => el.focus();
+const blurElement = (el: TextInput) => el.blur();
+const current = (ref: React.RefObject<TextInput>) => ref.current;
+
+interface Props {
+  compareWithCode?: string;
+  inactiveColor: string;
+  activeColor: string;
+  onFulfill: (code: PinString, isValid: boolean) => void;
+}
+
+interface State {
+  value: PinString;
```

Comment:
you should not use `PinString` for intermediate values, `PinString` is supposed to be the type of a valid PIN (you can cast the intermediate value to a `PinString` once validation passes)